lib/src/cli/commands/command_mixin.dart

```diff
@@ -12,20 +12,48 @@ mixin CommandMixin {
 Package reference can be one of:
 - directory path pointing to a package on disk
   (e.g. /path/to/package)
-- any package from pub 
+- any package from pub
   (e.g. pub://package_name/version)
 ''';
 
+  String copyDepthExplanation(String inputKind) {
+    return '''
+Integer that determines how many path levels up from $inputKind package reference should be copied along with the package itself.
+This option allows to copy bigger part of the packages tree to make required path dependencies available for 'pub get' execution.
+This number can not be greater then the count of components of the path given in the --$inputKind parameter.
+This parameter is ignored for published packages and absolute paths. Defaults to 0.
+ ''';
+  }
+
   /// prepares given [ref]. Depending on the type of ref this can include
   /// - copying the package to a temporary directory
   /// - running pub get
   /// If you use [analyze] with this result then it will take care to clean up
   /// everything (e.g. removing temp directory)
-  Future<PreparedPackageRef> prepare(PackageRef ref) async {
+  Future<PreparedPackageRef> prepare(PackageRef ref, {int? depth}) async {
     String sourceDir = '';
+    String? packageRelativePath;
     if (ref.isDirectoryPath) {
       stdout.writeln('Preparing ${ref.ref}');
       sourceDir = ref.ref;
+      if (depth != null && depth != 0 && !ref.isAbsolutePath) {
+        if (sourceDir.endsWith('/')) {
+          sourceDir = sourceDir.substring(0, sourceDir.length - 1);
+        }
+        int componentsCount =
+            sourceDir == '.' ? 0 : sourceDir.split('/').length;
+        if (componentsCount < depth) {
+          throw ArgumentError(
+              'Package path (${ref.ref}) components count ($componentsCount) is lower than package relative location depth ($depth)');
+        }
+        packageRelativePath =
+            sourceDir.split('/').sublist(componentsCount - depth).join('/');
+        if (packageRelativePath.split('/').any((element) => element == '..')) {
+          throw ArgumentError(
+              'Package path (${ref.ref}) should not have ".." as one of last $depth components (copy-depth).');
+        }
+        sourceDir = ref.ref + '/..' * depth;
+      }
```

Comment:
I would suggest to use `p.split()` and `p.join()` as it already handles the trailing path delimiter and works with `/` on Unix like systems as well as with `\` on Windows